fvm/environment/system_contracts.go

```diff
@@ -92,6 +92,34 @@ func ServiceAddress(chain flow.Chain) flow.Address {
 	return chain.ServiceAddress()
 }
 
+var verifyPayersBalanceForTransactionExecutionSpec = ContractFunctionSpec{
+	AddressFromChain: FlowFeesAddress,
+	LocationName:     systemcontracts.ContractNameFlowFees,
+	FunctionName:     systemcontracts.ContractServiceAccountFunction_verifyPayersBalanceForTransactionExecution,
+	ArgumentTypes: []sema.Type{
+		sema.AuthAccountType,
+		sema.UInt64Type,
+		sema.UInt64Type,
+	},
+}
+
+// CheckPayerBalanceAndGetMaxTxFees executes the verifyPayersBalanceForTransactionExecution
+// on the FlowFees account.
+func (sys *SystemContracts) CheckPayerBalanceAndGetMaxTxFees(
+	payer flow.Address,
+	inclusionEffort uint64,
+	maxExecutionEffort uint64,
+) (cadence.Value, error) {
+	return sys.Invoke(
```

Comment:
do you have profiling info about how much this extra overhead will do? wondering if it can be less than deductFees & checkStorage cadence calls.